autowebcompat/network.py

```diff
@@ -185,15 +187,20 @@ def create_inception_network(input_shape):
     return Model(input, x)
 
 
-def create_resnet50_network(input_shape):
-    base_model = ResNet50(input_shape=input_shape)
+def create_resnet50_network(input_shape, weights):
+    base_model = ResNet50(input_shape=input_shape, weights=weights)
     return Model(inputs=base_model.input, outputs=base_model.get_layer('flatten_1').output)
 
 
-def create(input_shape, network='vgglike', weights=None):
+def create(input_shape, network='vgglike', weights=None, builtin_weights=None):
     assert network in SUPPORTED_NETWORKS, '%s is an invalid network' % network
+    assert weights is None or builtin_weights is None, 'only one type of weights are allowed at once'
+
+    if builtin_weights:
+        assert network in SUPPORTED_NETWORKS_WITH_WEIGHTS, '%s does not have weights for %s ' % (network, builtin_weights)
+
```

Comment:
You could also assert that builtin_weights is in SUPPORTED_WEIGHTS.

Reply:
I think that is already asserted when I pass it as argparse using `choice=network.SUPPORTED_WEIGHTS`

Reply:
That's just the user interface. In the future there could be other things calling the library without going through the user interface, so it's good to program "defensively".